Model/Sync/Category/Processor/Main.php

```diff
@@ -409,4 +419,25 @@ public function checkForCollectionExistsError(DataObject $response): bool
     {
         return '409' == $response->getData('status');
     }
+
+    /**
+     * @return CategoryCollection<mixed>
+     * @throws LocalizedException
+     * @throws NoSuchEntityException
+     */
+    public function getStoreCategoryCollection()
+    {
+        /** @var \Magento\Store\Model\Store  $currentStore**/
```

Comment:
@manaftryzens 
Should it be removed?

Reply:
@manaftryzens I removed this line on the [test branch](https://github.com/YotpoLtd/magento2-module-core/pull/159), and merged to `master`.
Let me know if it was left for a purpose.

Reply:
Hi @EranAvidor ,

We need that line there. It is actually explicitly telling to PHPSTAN that $currentStore is an instance of \Magento\Store\Model\Store. 

If this line is not there, it will take $currentStore as a direct instance of Magento\Store\Model\StoreManagerInterface and throw the below warning
**Call to an undefined method Magento\Store\Api\Data\StoreInterface::getRootCategoryId().**

So we have kept that line there to resolve this warning